lib/phoenix/channel.ex

```diff
@@ -338,13 +338,39 @@ defmodule Phoenix.Channel do
     Server.push(transport_pid, topic, event, message, socket.serializer)
   end
 
+  @doc """
+  Replies asynchronously to a socket push.
+
+  Useful when you need to reply to a push that can't otherwise be handled using
+  the `{:reply, {status, payload}, socket}` return from your `handle_in`
+  callbacks. `reply/3` will be used in the rare cases you need to perform work in
+  another process and reply when finished by providing the push's `socket.ref`.
+
+  ## Examples
+
+      def handle_in("work", payload, socket) do
+        Worker.perform(payload, socket.ref)
+        {:noreply, socket}
+      end
+
+      def handle_info({:work_complete, result, ref}, socket) do
+        reply socket, ref, {:ok, result}
+        {:noreply, socket}
+      end
+
+  """
+  def reply(socket, ref, {status, payload}) do
```

Comment:
Analogous to GenServer.reply, maybe isn't the right word, but the idea is you generally call `reply` within the context of your channel server, and only pass around your `socket.ref`, like the example I layed out. The other option is to force users to provide `socket.transport_pid`, but then we are leaking a little bit the details. This works well for Tasks for example:

      def handle_in("work", params, socket) do
        ref = socket.ref
        Task.async(fn -> reply(socket, ref, {:ok, expensive_call(params) end)
        {:noreply, socket}
      end